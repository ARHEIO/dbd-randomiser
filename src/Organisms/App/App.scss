@import url('https://fonts.googleapis.com/css?family=Open+Sans|Quicksand');

html,
body {
  margin: 0;
  height: 100%;
  background-color: #1F2020;
}

.app-container {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100vh;

  .app {
    align-items: stretch;
    display: flex;
    flex-direction: column;
    margin: 0 auto;
    width: 100%;

    @media (min-width: 960px) {
      width: 800px;
    }
  }
}
