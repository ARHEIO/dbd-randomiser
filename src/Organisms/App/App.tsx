import React from 'react';
import './App.scss';

import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

import Landing from '../Landing/Landing';
import Killer from '../Killer/Killer';
import Survivor from '../Survivor/Survivor';

function App() {
  const appFinishedLoading = true;

  return (
    <div className="app-container">
      <header className="app-header">
        <div className="app-header_top-bar"></div>
        <div className="app-header_bottom-bar">
          <h1>Dead By Daylight Loadout Randomiser</h1>
        </div>
      </header>
      <section className="app">
        <Router
          basename={`/dbd-randomiser`}>
          {
            appFinishedLoading
            ? (
              <Switch>
                <Route path='/killer' component={Killer} />
                <Route path='/survivor' component={Survivor} />
                <Route path='/*' exact component={Landing} />
              </Switch>
              ) : <div></div>
            }
        </Router>
      </section>
      <footer>
        <p>Credit to icons goes to u/thatpikminguy on Reddit</p>
        <p>Grab the icon set yourself <a href="https://www.reddit.com/r/PerkByDaylight/comments/fhqoux/the_shitty_ui_pack_13_chains_of_hate_doctor/" target="_blank">on r/PerkByDaylight</a></p>
      </footer>
    </div>
  );
}

export default App;
