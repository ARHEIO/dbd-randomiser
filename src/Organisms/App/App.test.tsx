import React from 'react';
import { render } from '@testing-library/react';
import pretty from 'pretty';
import App from './App';

it('renders learn react link', () => {
  const { container } = render(<App />);
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
