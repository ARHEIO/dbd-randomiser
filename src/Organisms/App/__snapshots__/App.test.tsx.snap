// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
"<div class=\\"app-container\\">
  <header class=\\"app-header\\">
    <div class=\\"app-header_top-bar\\"></div>
    <div class=\\"app-header_bottom-bar\\">
      <h1>Dead By Daylight Loadout Randomiser</h1>
    </div>
  </header>
  <section class=\\"app\\">
    <div class=\\"landing-container\\"><a class=\\"link-button\\" href=\\"#/survivor\\"><img alt=\\"survivor\\" src=\\"https://dqr3pglopijar.cloudfront.net/assets/iconHelpLoading_survivor.png\\">
        <p>Randomise survivor build</p>
      </a><a class=\\"link-button\\" href=\\"#/killer\\"><img alt=\\"killer\\" src=\\"https://dqr3pglopijar.cloudfront.net/assets/iconHelpLoading_killer.png\\">
        <p>Randomise killer build</p>
      </a></div>
  </section>
  <footer class=\\"app-footer\\">
    <p>Credit to icons goes to u/thatpikminguy on Reddit. Grab the icon set yourself on<a href=\\"https://www.reddit.com/r/PerkByDaylight/comments/fhqoux/the_shitty_ui_pack_13_chains_of_hate_doctor/\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\">r/PerkByDaylight</a></p>
  </footer>
</div>"
`;
